packages/git/src/browser/git-navigable-list-widget.ts

```diff
@@ -99,32 +87,40 @@ export class GitBaseWidget<T extends { selected?: boolean }> extends VirtualWidg
         return result;
     }
 
-    protected onAfterAttach(msg: Message): void {
-        super.onAfterAttach(msg);
-        this.addKeyListener(this.node, Key.ARROW_LEFT, () => this.handleLeft());
-        this.addKeyListener(this.node, Key.ARROW_RIGHT, () => this.handleRight());
-        this.addKeyListener(this.node, Key.ARROW_UP, () => this.handleUp());
-        this.addKeyListener(this.node, Key.ARROW_DOWN, () => this.handleDown());
-        this.addKeyListener(this.node, Key.ENTER, () => this.handleEnter());
+    protected handleListKeyDown(e: KeyboardEvent): void {
+        const kc = KeyCode.createKeyCode(e);
+        if (kc.equals(KeyCode.createKeyCode({ first: Key.ARROW_LEFT }))) {
+            this.navigateLeft();
+        } else if (kc.equals(KeyCode.createKeyCode({ first: Key.ARROW_RIGHT }))) {
+            this.navigateRight();
+        } else if (kc.equals(KeyCode.createKeyCode({ first: Key.ARROW_UP }))) {
+            this.navigateUp();
+        } else if (kc.equals(KeyCode.createKeyCode({ first: Key.ARROW_DOWN }))) {
+            this.navigateDown();
+        } else if (kc.equals(KeyCode.createKeyCode({ first: Key.ENTER }))) {
+            this.handleListEnter();
+        }
+        e.stopPropagation();
```

Comment:
You should not stop propagation for all keys. I think you should rollback to `this.addKeyListener`. It ensures it.